### PR TITLE
Replace gsub with delete_suffix.

### PR DESCRIPTION
**Pull request by pranavbabu**
This PR aims to enhance the codebase of  project by replacing usage of `gsub` method with the more appropriate `delete_suffix` method where applicable. This change not only improves the clarity and readability of the code but also provides potential performance benefits in certain scenarios.

### CODE REVIEW

**Review by gazayas — APPROVED** [review]

**Comment by gazayas** [comment]
Thanks for this @pranavbabu!

The tests are failing because we updated our tests in the starter repository recently, and the changes haven't been reflected in `bullet_train-core` yet. Please wait until #773 is merged, and then you can merge `main` into this branch to get it working.

**Comment by jagthedrummer** [comment]
@pranavbabu could you rebase this branch against `main`? I think we should be able to merge it after that.

**Pull request closed without merging**
